db_service/vcap.go

```diff
@@ -0,0 +1,98 @@
+package db_service
+
+import (
+	"code.cloudfoundry.org/lager"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/viper"
+	"net/url"
+	"os"
+)
+
+type VcapService struct {
+	BindingName  string            `json:"binding_name"`  // The name assigned to the service binding by the user.
+	InstanceName string            `json:"instance_name"` // The name assigned to the service instance by the user.
+	Name         string            `json:"name"`          // The binding_name if it exists; otherwise the instance_name.
+	Label        string            `json:"label"`         // The name of the service offering.
+	Tags         []string          `json:"tags"`          // An array of strings an app can use to identify a service instance.
+	Plan         string            `json:"plan"`          // The service plan selected when the service instance was created.
+	Credentials  map[string]string `json:"credentials"`   // The service-specific credentials needed to access the service instance.
+}
+
+func useVcapServices(logger lager.Logger) error {
+	vcapData, vcapExists := os.LookupEnv("VCAP_SERVICES")
+	if vcapExists {
+		vcapService, err := parseVcapServices(vcapData, logger)
+		if err != nil {
+			return fmt.Errorf("Error parsing VCAP_SERVICES: %s", err)
+		}
+
+		// if URI is supplied, we should parse it to fill any missing fields
+		u, err := url.Parse(vcapService.Credentials["uri"])
+		if err != nil {
+			return fmt.Errorf("Error parsing VCAP_SERVICES credentials URI: %s", err)
+		}
+
+		logger.Info("Using MySQL database injected via VCAP_SERVICES environment variable")
+		viper.Set(dbPathProp, u.Path)
+		viper.Set(dbTypeProp, DbTypeMysql)
+		viper.Set(dbHostProp, vcapService.Credentials["host"])
+		viper.Set(dbUserProp, vcapService.Credentials["Username"])
+		viper.Set(dbPassProp, vcapService.Credentials["Password"])
+		viper.Set(dbNameProp, vcapService.Credentials["database_name"])
```

Comment:
let's also add a check for just Credentials["name"] because that's what Pivotal uses in theirs https://docs.pivotal.io/p-mysql/2-6/use.html#vcap

Reply:
are you thinking:
if `vcapServices.Credentials["database_name"]` is null {
check for `vcapServices.credentials["name"]`?
}

Reply:
Yeah, something like that; the same could go for uppercase/lowercase password, and username.

Reply:
But maybe also a third option from the parsed URI (if it's not blank)